src/main.cpp

```diff
@@ -179,10 +220,17 @@ MODLOADER_FUNC void modloader_accept_unity_handle([[maybe_unused]] JNIEnv* env,
   }
   void* il2cpp_init = dlsym(modloader_libil2cpp_handle, "il2cpp_init");
   LOG_DEBUG("Found il2cpp_init at: {}", fmt::ptr(il2cpp_init));
-  modloader_unity_handle = unityHandle;
   // Hook il2cpp_init to redirect to us after loading, and load normal mods
   // TODO: We KNOW it has enough space for us, but we could technically double check this too.
   install_load_hook(reinterpret_cast<uint32_t*>(il2cpp_init));
+
+  void* unity_playerstartfirstscene = dlsym(modloader_unity_handle, "PlayerStartFirstScene");
+  // Hook PlayerStartFirstScene so we can load in mods (aka late mods)
+  // this method is ran to load the first scene of the game (null -> QuestInit for beat saber), and thus after this we are sure unity is initialized
+  // TODO: should we hook UnityInitApplication instead? there is no guarantee that things would work correctly that early, since no scene would be loaded yet
```

Comment:
Feedback request: we could hook UnityInitApplication instead, but at that point no scene exists yet so GameObjects created at that point couldn't possibly belong to *any* scene.